lib/model/user.dart

```diff
@@ -1,16 +1,24 @@
+import 'package:flutter/cupertino.dart';
+
 class User {
-  final String id;
-  final String username;
+  String id;
+  String email;
+  String userName;
+  String photoURL;
 
   User({
     this.id,
-    this.username,
+    this.email,
+    this.userName,
+    this.photoURL,
   });
 
-  User copyWith({String id, String username}) {
+  User copyWith({@required String id, String email, String userName, String photoURL}) {
     return User(
       id: id ?? this.id,
-      username: username ?? this.username,
+      email: email ?? this.email,
+      userName: userName ?? this.userName,
+      photoURL: photoURL ?? "_",
```

Comment:
Unless you have a good reason for this.

Reply:
Default string for photoURL should be something that we can attach an image to...I've found that it was easier to place a simple if statement to attach the default image to...if (doc.data()['photoURL') == _ {place default image}

Reply:
Can we not check if the photoURL is null before placing the default image? Either way the copyWith function doesn't seem like the correct place to enforce it, it will overwrite the user's photoURL every chance it gets.